src/maintainer/infrastructure.rst

```diff
@@ -281,35 +281,134 @@ Compilers and Runtimes
+* [C, C++] Required feedstocks:
+  `llvmdev <https://github.com/conda-forge/llvmdev-feedstock>`_,
+  `clangdev <https://github.com/conda-forge/clangdev-feedstock>`_,
+  `compiler-rt <https://github.com/conda-forge/compiler-rt-feedstock>`_,
+  `libcxx <https://github.com/conda-forge/libcxx-feedstock>`_,
+  `openmp <https://github.com/conda-forge/openmp-feedstock>`_,
+  `lld <https://github.com/conda-forge/lld-feedstock>`_,
+  `cctools <https://github.com/conda-forge/cctools-and-ld64-feedstock>`_
+* [Fortran] Activation: https://github.com/conda-forge/gfortran_osx-64-feedstock/
+* [Fortran] Implementation: https://github.com/conda-forge/gfortran_impl_osx-64-feedstock/
+
+Windows (MSVC):
+
+* [C, C++] Activation: https://github.com/conda-forge/vc-feedstock
+  (we cannot redistribute the actual MSVC compilers due to licensing constraints)
+* [Fortran] Activation & Implementation: https://github.com/conda-forge/flang-feedstock
+
+There exists an alternative, MinGW-based, compiler stack on windows, which is available
+with a ``m2w64_`` prefix (e.g. ``{{ compiler('m2w64_c') }}``). However, it is falling out
+of use now that most projects will natively support compilation also with MSVC, in addition
+to several complications arising from mixing compiler stacks.
+
+Additionally, there is a possibility to use clang as a compiler on Linux & Windows:
+
+* Activation (Linux): https://github.com/conda-forge/clang-compiler-activation-feedstock/
+* Activation (Windows): https://github.com/conda-forge/clang-win-activation-feedstock/
+
+Aside from the main C/C++/Fortran compilers, these are the feedstocks for the other compilers:
+
+* [CUDA] https://github.com/conda-forge/nvcc-feedstock (CUDA infra currently being overhauled)
+* [Rust] `Activation <https://github.com/conda-forge/rust-activation-feedstock>`__
+  and `Implementation <https://github.com/conda-forge/rust-feedstock>`__
+* [Go] `Activation <https://github.com/conda-forge/go-activation-feedstock>`__
+  and `Implementation <https://github.com/conda-forge/go-feedstock>`__
+
+To upgrade the compiler version of our default compilers in the global pinning for
+Linux or OSX, ensure that the respective above-mentioned feedstocks have been rebuilt
+for the new major version, that all interrelated versions are lifted at the same time,
+and obviously that the compilers work (e.g. by testing them on some feedstocks by
+specifying the new version through the feedstock-local `conda_build_config.yaml`).
+You should also check the compiler release notes for warnings about ABI incompatibilities,
+and mention any such notices in the discussion about the upgrade.
+
+For Windows, we stay on older compilers for longer, because using a newer toolchain
+would force everyone wanting to locally develop with conda-forge artefacts to use
+a toolchain that's at least as new. A more in-depth explanation can be found
+`here <https://github.com/conda-forge/conda-forge.github.io/issues/1732>`_.
```

Comment:
It's usually better to have a link text that requires no context to understand its meaning (see [this accessibility guidelines section](https://usability.yale.edu/web-accessibility/articles/links#link-text) for details).

So this could be rephrased like:

```suggestion
a toolchain that's at least as new. You can find more details about this topic in this
`issue about updating to the vc142 toolchain <https://github.com/conda-forge/conda-forge.github.io/issues/1732>`_.
```

(or something similar).